#! /bin/sh
patch -p0 -f $* < $0
exit $?

[g10]
2009-09-28  Werner Koch  <wk@g10code.com>

       * trustdb.c (get_validity_info): Take care of a NULL PK.  Fixes
       bug#1138.
       (get_validity_string): Ditto.


--- g10/trustdb.c.orig  2008-12-09 09:26:37.000000000 +0100
+++ g10/trustdb.c       2009-09-28 11:06:28.000000000 +0200
@@ -1157,6 +1157,9 @@
 {
     int trustlevel;
 
+    if (!pk)
+        return '?';  /* Just in case a NULL PK is passed.  */
+
     trustlevel = get_validity (pk, uid);
     if( trustlevel & TRUST_FLAG_REVOKED )
        return 'r';
@@ -1168,6 +1171,9 @@
 {
   int trustlevel;
 
+  if (!pk)
+    return "err";  /* Just in case a NULL PK is passed.  */
+
   trustlevel = get_validity (pk, uid);
   if( trustlevel & TRUST_FLAG_REVOKED )
     return _("revoked");


